File: backend/database/redis.go
package database

import (
	"context"
	"log"

	"lifecompass-backend/config"

	"github.com/redis/go-redis/v9"
)

var RDB *redis.Client
var Ctx = context.Background()

func ConnectRedis(cfg *config.Config) {
	opts, err := redis.ParseURL(cfg.RedisURL)
	if err != nil {
		log.Fatalf("Failed to parse Redis URL: %v", err)
	}

	RDB = redis.NewClient(opts)

	if err := RDB.Ping(Ctx).Err(); err != nil {
		log.Printf("Warning: Failed to connect to Redis: %v. Caching will be disabled.", err)
		RDB = nil
		return
	}

	log.Println("Redis connected successfully")
}

File: backend-express/database/redis.js
const { createClient } = require('redis');
const config = require('../config/config');

let redisClient = null;

async function connectRedis() {
  if (!config.redisUrl) {
    console.log('No REDIS_URL configured. Caching will be disabled.');
    return null;
  }

  try {
    const client = createClient({
      url: config.redisUrl,
    });

    client.on('error', (err) => {
      // Log redis errors, but do not crash the app
      console.warn('Redis Client Error:', err.message);
    });

    await client.connect();
    console.log('Redis connected successfully');
    redisClient = client;
    return redisClient;
  } catch (err) {
    console.warn(`Warning: Failed to connect to Redis: ${err.message}. Caching will be disabled.`);
    redisClient = null;
    return null;
  }
}

function getRedisClient() {
  return redisClient;
}

module.exports = {
  connectRedis,
  getRedisClient,
};
